src/main/resources/db/migration/R__functions.sql

```diff
@@ -0,0 +1,335 @@
+BEGIN;
+
+-- Some utility functions first
+--
+CREATE OR REPLACE FUNCTION hash_bigint(TEXT) RETURNS BIGINT AS
+$$
+SELECT ('x' || substr(md5($1), 1, 16)) :: BIT(64) :: BIGINT;
```

Comment:
MD5 triggers people, can you replace this by SHA256?

Reply:
why? :)
we are hashing the string to get a number out of it to use as a lock key, 
so one collision in 10^N cases is not a big deal. It can be any relatively 
okay-ish string -> int8 hashing.

Reply:
Using MD5 is never the right choice nowadays (even if it is still secure, e.g. HMAC-MD5). If you need a cryptographic hash function: SHA256. If you don't need a cryptographic hash function (and the speed matters), pick one that is not.

Reply:
I don't need cryptographic hash function for sure.